### Eksploracyjna analiza danych
Zbiór danych który analizujemy to Loan_Default.csv, zawiera on dane o kredytobiorcach pewnego banku.

Importowanie bibliotek:

In [30]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

Wczytanie zbioru danych:

In [31]:
df = pd.read_csv("../../data/Loan_Default.csv")

Rozmiar zbioru danych:

In [32]:
df.shape

(148670, 34)

Jak widać w naszym zbiorze danych mamy 148670 rekordów z 34 zmiennymi.<br>
Występują w nim zarówno zmienne numeryczne, jak i kategoryczne:

In [33]:
numeric_variables = df.select_dtypes(include='number').columns.to_list()
categorical_variables = df.select_dtypes(exclude='number').columns.to_list()
print(f'Ilość zmiennych numerycznych: {len(numeric_variables)}')
print(f'Zmienne numeryczne: {numeric_variables}')
print(f'Ilość zmiennych kategorycznych: {len(categorical_variables)}')
print(f'Zmienne kategoryczne: {categorical_variables}')

Ilość zmiennych numerycznych: 13
Zmienne numeryczne: ['ID', 'year', 'loan_amount', 'rate_of_interest', 'Interest_rate_spread', 'Upfront_charges', 'term', 'property_value', 'income', 'Credit_Score', 'LTV', 'Status', 'dtir1']
Ilość zmiennych kategorycznych: 21
Zmienne kategoryczne: ['loan_limit', 'Gender', 'approv_in_adv', 'loan_type', 'loan_purpose', 'Credit_Worthiness', 'open_credit', 'business_or_commercial', 'Neg_ammortization', 'interest_only', 'lump_sum_payment', 'construction_type', 'occupancy_type', 'Secured_by', 'total_units', 'credit_type', 'co-applicant_credit_type', 'age', 'submission_of_application', 'Region', 'Security_Type']


W niektórych kolumnach występują braki danych:

In [34]:
na_columns = df.columns[df.isna().any()].to_list()
print(f"Kolumny z brakami danych: {na_columns}")
print(f"Ilość kolumn z brakami danych: {len(na_columns)}")

Kolumny z brakami danych: ['loan_limit', 'approv_in_adv', 'loan_purpose', 'rate_of_interest', 'Interest_rate_spread', 'Upfront_charges', 'term', 'Neg_ammortization', 'property_value', 'income', 'age', 'submission_of_application', 'LTV', 'dtir1']
Ilość kolumn z brakami danych: 14
